tools/sigh.ts

```diff
@@ -421,6 +443,94 @@ function link(srcFiles: Iterable<string>): boolean {
   return success;
 }
 
+// Config for building wasm modules. Keys are the target filename for the module.
+type WasmConfig = {
+  [key: string]: {
+    manifest: string,
+    src: string[]
+  }
+};
+
+// TODO: detect old headers/wasm modules/manifests in cleanObsolete()
+function buildWasmModule(configFile: string): boolean {
+  const srcDir = path.dirname(configFile);
+  const buildDir = srcDir.replace(/^src\//, 'build/');
+  const json = JSON.parse(fs.readFileSync(configFile, 'utf-8')) as WasmConfig;
+  let success = true;
+  for (const [name, cfg] of Object.entries(json)) {
+    // TODO: fix arcs.h so more than one source file can be compiled into a module
+    if (cfg.src.length !== 1) {
+      throw new Error(`wasm modules must specify exactly one source file (${configFile})`);
+    }
+
+    const manifestPath = path.join(srcDir, cfg.manifest);
```

Comment:
`.manifest` could be undefined or null here, not sure if path.join handles that appropriately.

Reply:
I haven't been particularly diligent with error handling in this - the inputs (wasm.json) are our own files.

Reply:
Following on: I just tested this and `join() ` throws an error on null and undefined, which is fine.